lib/parallel_tests/tasks.rb

```diff
@@ -117,16 +117,18 @@ def parse_args(args)
   desc "Update test databases by dumping and loading --> parallel:prepare[num_cpus]"
   task(:prepare, [:count]) do |_,args|
     ParallelTests::Tasks.check_for_pending_migrations
-    if defined?(ActiveRecord) && ActiveRecord::Base.schema_format == :ruby
-      # dump then load in parallel
-      Rake::Task['db:schema:dump'].invoke
-      Rake::Task['parallel:load_schema'].invoke(args[:count])
-    else
-      # there is no separate dump / load for schema_format :sql -> do it safe and slow
-      args = args.to_hash.merge(:non_parallel => true) # normal merge returns nil
-      taskname = Rake::Task.task_defined?('db:test:prepare') ? 'db:test:prepare' : 'app:db:test:prepare'
-      ParallelTests::Tasks.run_in_parallel("#{ParallelTests::Tasks.rake_bin} #{taskname}", args)
+    # dump schema/structure then load in parallel
+    unless defined?(ActiveRecord::Base)
+      raise "ActiveRecord is required for rake parallel:prepare!"
```

Comment:
can we keep the original where it just called `db:test:prepare` instead of crashing ?
... could add a `warn "Using slower db:test:prepare because ActiveRecord was not loaded"` ...

Reply:
Ahh right, yes that makes sense! I have re-added that code so it just calls `db:test:prepare` if `ActiveRecord::Base` is not defined. (Also please ping me if any of these changes cause any problems for other users!)